client/store.go

```diff
@@ -200,78 +199,24 @@ func (c *Client) PullBatch(ctx context.Context, recordRefs []*corev1.RecordRef)
 	return records, nil
 }
 
-// PushWithOptions sends a record with optional OCI artifacts like signatures to the store.
-func (c *Client) PushWithOptions(ctx context.Context, record *corev1.Record, signature *signv1.Signature) (*storev1.PushWithOptionsResponse, error) {
-	// Create streaming client
-	stream, err := c.StoreServiceClient.PushWithOptions(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create push with options stream: %w", err)
-	}
-
-	// Create push options
-	options := &storev1.PushOptions{
-		Signature: signature,
-	}
-
-	// Create request
-	request := &storev1.PushWithOptionsRequest{
-		Record:  record,
-		Options: options,
-	}
-
-	// Send request
-	if err := stream.Send(request); err != nil {
-		return nil, fmt.Errorf("failed to send push with options request: %w", err)
-	}
-
-	// Close send stream
-	if err := stream.CloseSend(); err != nil {
-		return nil, fmt.Errorf("failed to close send stream: %w", err)
-	}
-
-	// Receive response
-	response, err := stream.Recv()
-	if err != nil {
-		return nil, fmt.Errorf("failed to receive push with options response: %w", err)
-	}
-
-	return response, nil
-}
-
-// PullWithOptions retrieves a record along with its associated OCI artifacts.
-func (c *Client) PullWithOptions(ctx context.Context, recordRef *corev1.RecordRef, includeSignature bool) (*storev1.PullWithOptionsResponse, error) {
-	// Create streaming client
-	stream, err := c.StoreServiceClient.PullWithOptions(ctx)
+// PushWithSigning pushes a record and optionally signs it using the separate signing flow.
+func (c *Client) PushWithSigning(ctx context.Context, record *corev1.Record, sign bool, signatureProvider *signv1.SignRequestProvider) (*corev1.RecordRef, error) {
```

Comment:
do we really need this? seems redundant

Reply:
we already have all the primitives we need, i dont think we need to simply further.

instead, we would need the `PushRefferer` and `PullReferrer` functions